text/0000-location.md

```diff
@@ -0,0 +1,409 @@
+digit.
+
+### Referencing Locations
+
+Locations are uniquely referenced by their location_id and location_namespace.
+For GS1, locations are referenced by the GLN identifier.
+
+**Note: For the remainder of this document, �address� will refer to a 
+Grid/Transact address as opposed to a Street Address.**
+
+### Location Addressing in the Merkle-Radix State System
+
+In order to uniquely locate GS1 locations in the Merkle-Radix state system, an 
+address must be constructed which identifies the storage location of the 
+Location representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix �621dee�.
+Locations are further prefixed under the Grid namespace with reserved 
+enumerations of �04� (�00�, �01�, �02� and �03� being reserved for other 
+purposes) indicating �Locations� and an additional �01� indicating �GS1 
+Locations�.
+
+Therefore, all addresses starting with �621dee� + �04� are Grid locations, and 
+more specifically, all addresses starting with �621dee� + �04� + �01� are Grid 
+GS1 Locations identified by a GLN.
+
+The GLN format consists of a 13-digit �numeric string� that follows a similar 
+structure to GTIN-13. The GLN structure first includes a GS1 Company Prefix 
+which is 7-10 digits in length and is assigned by a GS1 Member Organization. A 
+Location Reference number follows which is 2-5 digits in length and is 
+allocated by the company to a location or party. Lastly, a single digit Check 
+Digit is calculated and applied according to a GS1 algorithm.
+
+![GLN Data Format](https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSFo3J5QULgkCWyIH3yeddRPx1H5aQusLRRq47y1n2FqTshGxs%3Ax-raw-image%3A%2F%2F%2Ff4fc819afc288cee1948ca0d40d76c3bcd56a6d270fcfeeac80fd25784eb6071&usqp=CAU)
+
+After the 10-hex-characters that are consumed by the grid namespace prefix, the 
+location, and GS1 prefixes, there are 60 hex characters remaining in the address.
+The 13 digits of the GLN can be left padded with 45-hex-character zeroes and 
+right padded with 2-hex-character zeroes to accommodate potential future storage
+associated with the GS1 Location representation, for example:
+
+    �621dee� + �04� + �01� + �000000000000000000000000000000000000000000000� +
+    13-character �numeric string� location_id + �00� // location_id == GLN
+
+A full GS1 Location address (for example purposes) would therefore be:
+
+    �621dee0401000000000000000000000000000000000000000000000123456789012800�
+
+
+## Transaction Payload and Execution
+[transaction payload and execution]: "transactionpayloadandexecution"
+
+### Location PayLoad Transaction
+
+LocationPayload contains an action enum and the associated action payload. This 
+allows for the action payload to be dispatched to the appropriate logic.
+
+Only the defined actions are available and only one action payload should be 
+defined in the LocationPayload.
+
+    message LocationPayload {        
+
+     enum Actions {
```

Comment:
Indentation is incorrect on this line and following lines.

Reply:
Is there published guidance I can reference to correct the indentation? Or a simple rule of thumb you can share? I just tabbed in until the text turned grey.

Reply:
Each block (defined by {}) is intended with 4 additional spaces. so this should look like:

```
    message LocationPayload {        
        enum Actions {
            UNSET_ACTION = 0;
            LOCATION_CREATE = 1;
            LOCATION_UPDATE = 2;
            LOCATION_DELETE = 3;
        }

        Action action = 1;

        // Approximately when transaction was submitted, as a Unix UTC timestamp
        uint64 timestamp = 2;

        LocationCreateAction location_create = 3;
        LocationUpdateAction location_update = 4;
        LocationDeleteAction location_delete = 5;
    }
```